src/System.Runtime/ref/System.Runtime.cs

```diff
@@ -6139,7 +6139,7 @@ public class TypeDelegator : System.Reflection.TypeInfo
     public delegate bool TypeFilter(Type m, Object filterCriteria);
     public abstract partial class TypeInfo : System.Type, System.Reflection.IReflectableType
     {
-        internal TypeInfo() { }
+        protected TypeInfo() { }
```

Comment:
alright! it would of course also be possible to move those 3 classes into coreclr then the would be able to call the ctor of TypeInfo as it internal there.